symengine/mp_class.h

```diff
@@ -413,6 +431,11 @@ inline double mp_get_d(const piranha::integer &i)
     return mpz_get_d(i.get_mpz_view());
 }
 
+inline void mp_set_d(piranha::integer &i, double a)
+{
+    return mpz_set_d(i.get_mpz_view(), a);
```

Comment:
There should be a way in piranha to do this. `i = a` or `i = piranha::integer(a)` might work.